relay-server/src/actors/upstream.rs

```diff
@@ -402,6 +402,7 @@ impl UpstreamRelay {
                     .connect_timeout(config.http_connection_timeout())
```

Comment:
I just noticed that you have `rt-multi-thread` enabled. This means that the runtime you create here is multi threaded, as opposed to the single thread that the actix client runs in. Can you restrict this to a single thread and test throughput? 

While it makes sense to eventually multi-thread, let's please stick to the proven behavior so that upstream connections to not compete with event processors for resources.

Reply:
In Symbolicator we do [this](https://github.com/getsentry/symbolicator/blob/f4351ab642071eebb33f6a308e0bc9aa927aca4e/src/app.rs#L39-L44), although in Relay we should be able to run with just the `current_thread` scheduler, too.

Reply:
You're right. I wanted to preserve tokio 0.x behavior but actually the default ::new() was single-threaded in 0.x. So I committed your suggestion.

I cannot explain it but current_thread yields 503 in the healthcheck in tests. I believe it blocks the wrong thread for too long.

Based on that I believe it would be safest to defer merge of this PR until after the OSS release. The OSS release should get by fine with getaddrinfo DNS resolution and old tokio.

Reply:
reqwest internally likely needs blocking I/O, so use `rt-multi-thread` and spawn with a single worker thread. This works in symbolicator and will preserve behavior.